generator/default.p9k

```diff
@@ -434,15 +448,21 @@ function __p9k_render() {
   # to avoid double expansion of the contents of the PROMPT.
   typeset -gAh __p9k_unsafe=()
 
+  local segment raw_segment
   # Process Cache
-  for data in "${(Oa@v)__p9k_segment_cache}"; do
-    [[ -z "${data}" ]] && continue
-
-    local -a segment_meta=("${(@s:·|·:)data}")
-    [[ "${segment_meta[7]}" == "false" ]] && continue # Segment should not be printed
-
-    local alignment="${segment_meta[2]}"
-    __p9k_${alignment}_prompt_segment "${segment_meta[1]}" "${segment_meta[3]}" "${segment_meta[4]}" "${segment_meta[5]}" "${segment_meta[6]}"
+  for alignment in left right; do
+    local segments_var="P9K_${(U)alignment}_PROMPT_ELEMENTS"
+    for raw_segment in ${(P)segments_var}; do
```

Comment:
Usually we don't need to know the segments name here. All we should need is the alignment and index, because we want to process the segments in that order.
You added another loop here to find the right segment by its name. This also has the limitation of having each segment only once per left/right prompt. I know this limitation is quite artificial, but the other way is faster and more flexible..